### PR TITLE
block_hotplug: Update function unplug device

### PR DESCRIPTION
**Pull request by YongxueHong**
Check if device is unplugged with getting qmp "DEVICE_DELETED" event.

Signed-off-by: Yongxue Hong <yhong@redhat.com>

ID: 1541869

### CODE REVIEW

**Comment by YongxueHong** [comment]
(workspace) [root@ibm-p8-kvm-02-qe staf-kvm-devel]# python ConfigTest.py --testcase=block_hotplug --driveformat=virtio_scsi --imageformat=qcow2 --nicmodel=virtio_net --guestname=RHEL.7.5 --platform=ppc64le --mem=8192 --vcpu=8 --verbose=no --clone=no
 (001/120) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.ppc64le.io-github-autotest-qemu.block_hotplug.block_virtio.fmt_qcow2.with_plug.one_pci: PASS (114.52 s)
 (002/120) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.ppc64le.io-github-autotest-qemu.block_hotplug.block_virtio.fmt_qcow2.with_plug.multi_pci: PASS (163.43 s)
...
 (119/120) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.ppc64le.io-github-autotest-qemu.block_hotplug.data_plane.block_scsi.fmt_qcow2.with_plug.with_system_reset.one_pci: PASS (275.57 s)
 (120/120) smp_8.8192m.repeat1.run_test.Host_RHEL.m7.u6.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.5.ppc64le.io-github-autotest-qemu.block_hotplug.data_plane.block_scsi.fmt_raw.with_plug.with_system_reset.one_pci: PASS (331.09 s)
RESULTS    : PASS 120 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB TIME   : 85710.71 s

File: qemu/tests/block_hotplug.py [hunk 1]
```diff
@@ -64,6 +65,22 @@ def unplug_device(vm, get_disk_cmd, device):
...
+                              % (params.get('event_name'),
+                                 vm.monitor.get_event(params.get('event_name'))))
+                vm.monitor.clear_event(params.get('event_name'))
+                event_acquired = True
+                break
+            time.sleep(float(params.get('check_interval_timeout')))
+        if not event_acquired:
+            test.fail("Can not get event \"%s\" under %s."
+                      % (params.get('event_name'), params.get('event_timeout')))
+
```
Comment (peixiu) [inline qemu/tests/block_hotplug.py]: Hi yhong, these codes looks as a part of verify device unplugged, if add them to verify_unplug() is better?
and for title word "fuction" miss letter "n".

Comment (YongxueHong) [reply qemu/tests/block_hotplug.py]: I don't think so, since here just get a deleted event info, not a action for verifying.

**Comment by aliang123** [comment]
For rhel guest: Acked-by: Aihua Liang<aliang@redhat.com>  
PASS 1-smp_4.1024m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.block_hotplug.block_virtio.fmt_raw.with_plug.one_pci
PASS 2-smp_4.1024m.repeat1.run_test.Host_RHEL.m7.u4.qcow2.virtio_blk.up.virtio_net.Guest.RHEL.7.5.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_qcow2.without_plug.multi_pci

**Comment by peixiu** [comment]
Tested on win10-64 guest:
 (1/2) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_raw.with_plug.multi_pci: PASS (281.58 s)
 (2/2) smp_4.2048m.repeat1.run_test.Host_RHEL.m7.u5.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.block_hotplug.block_scsi.fmt_raw.with_plug.multi_pci: PASS (260.28 s)

Acked-by: Peixiu Hou phou@redhat.com

File: qemu/tests/block_hotplug.py [hunk 2]
```diff
@@ -64,6 +64,20 @@ def unplug_device(vm, get_disk_cmd, device):
...
         device.unplug(vm.monitor)
+        event_status = utils_misc.wait_for(
+            lambda: vm.monitor.get_event(params.get('event_name')),
+            params.get('event_timeout'), 0.0,
+            float(params.get('check_interval')))
+
+        if not event_status:
+            test.fail("Can not get event \"%s\" under %s."
+                      % (params.get('event_name'), params.get('event_timeout')))
+        else:
```
Comment (hereischen) [inline qemu/tests/block_hotplug.py]: you don't need `else` here.
I think it maybe better to assign `params.get('event_name')` to a variable,say `event_name` , so you only need to get it once. 
Same to `params.get('event_timeout')`

**Comment by YongxueHong** [comment]
Refer to https://github.com/avocado-framework/avocado-vt/pull/1558.
So close this request.

**Pull request closed without merging**